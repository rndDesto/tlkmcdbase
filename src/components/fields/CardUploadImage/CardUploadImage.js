import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function Upload(props) {
  const { className, input, inputProps, data, label, helper, meta } = props;
  const { dirty, error, touched } = meta;
  const classes =   [
    styles.root,
    !!data || styles.placeholder,
    !!error && (dirty || touched) && styles.error,
    className
  ].filter(Boolean).join(' ');
  return (
    <div className={classes}>
      <label>{label}</label>
      <div className={classes}>
        {data ? <img alt="icon image" className={styles.preview} src={data} /> :
          <img alt="icon image" className={styles.icon} src="../../../assets/ic-default-image.svg" />
        }
        <label>
          {data ? <p>Edit</p>  : <p>Upload</p>}
          <input  id={input.name} {...input} {...inputProps} type="file"/>
        </label>
      </div>
      {!!error && (dirty || touched) && <small className={styles.error}>{error}</small>}
      <span>{helper}</span>
    </div>
  );
}
Upload.defaultProps = {
  className: '',
  data:'',
  helper: '',
  input: {},
  inputProps: {},
  label: '',
  meta:{},
};

Upload.propTypes = {
  className: PropTypes.string,
  data: PropTypes.string,
  helper: PropTypes.string,
  input: PropTypes.object,
  inputProps: PropTypes.object,
  label: PropTypes.string,
  meta: PropTypes.object,
};
