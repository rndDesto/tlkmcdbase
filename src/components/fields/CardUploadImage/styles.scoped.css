.root > label {
  color: var(--palette-black-20);
  font-size: 0.875rem;
  line-height: 1rem;
  font-weight: bold;
}

.root > div > label {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #CACCCF;
  width: 11.625rem;
  height: 2.5rem;
  cursor: pointer;
  border: 0.063rem solid #E8E8E8;
  z-index: 10;
}

.root > div > label > input[type="file"] {
  display: none;
}

.root > div > label > p {
  padding: 8px;
}

.root > div > label > div {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 2.5rem;
  height: 2.5rem;
  background-color: #cacccf;
  cursor: pointer;
  border-radius: 0px 3px 3px 0px;
}

.root > div > .icon {
  width: 2.375rem;
  height: 1.938rem;
  margin: 1.875rem 4.625rem;
}

.root > div > .preview {
  width: 100%;
  height: 6.5rem;
  object-fit: cover;
}

.root > div {
  border: 0.063rem solid #e8e8e8;
  border-radius: 0.188rem;
  box-shadow: none;
  color: var(--palette-black-10);
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  font-size: 0.875rem;
  font-weight: bold;
  margin-bottom: 0.5rem;
  margin-top: 0.625rem;
  width: 11.625rem;
  height: 8.75rem;
  transition: border 0.25s;
  background-color: #A0A4A8;
}

.root > div > label > p:nth-child(2) {
  align-items: center;
  background: red;
  border-radius: 3px;
  color: var(--palette-common-white);
  display: flex;
  justify-content: center;
  padding: 7px 16px;
  width: fit-content;
}

.root.placeholder > label > p:first-child {
  color: #25282B;
  display: flex;
  align-items: center;
}

.root > .error {
  border-color: var(--palette-textfield-error-main);
}

.root + small.error {
  color: var(--palette-textfield-error-main);
  display: block;
  font-size: 0.75rem;
}
