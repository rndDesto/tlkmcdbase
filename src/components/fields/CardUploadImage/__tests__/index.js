import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import CardUploadImage from '../CardUploadImage';

describe('src/components/elements/CardUploadImage', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<CardUploadImage />);
    expect(tree).toMatchSnapshot();
  });
});
