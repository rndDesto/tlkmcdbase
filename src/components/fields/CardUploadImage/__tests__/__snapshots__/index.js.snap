// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/CardUploadImage render 1`] = `
<div
  className=""
>
  <label>
    
  </label>
  <div
    className=""
  >
    <img
      alt="icon image"
      src="../../../assets/ic-default-image.svg"
    />
    <label>
      <p>
        Upload
      </p>
      <input
        type="file"
      />
    </label>
  </div>
  <span>
    
  </span>
</div>
`;
